platform/windows/detect.py

```diff
@@ -666,12 +666,12 @@ def get_ar_version(env):
         print_warning("Couldn't check version of `ar`.")
         return ret
 
-    match = re.search(r"GNU ar \(GNU Binutils\) (\d+)\.(\d+)(:?\.(\d+))?", output)
+    match = re.search(r"GNU ar \(GNU Binutils\) (\d+)\.(\d+)(:?\.)?(:?\d+)?", output)
```

Comment:
This looks weird to me. I feel this was meant to be a non capturing group, but the syntax is wrong, it should be `(?:\.`, not `(:?\.`.
```suggestion
    match = re.search(r"GNU ar \(GNU Binutils\) (\d+)\.(\d+)(?:\.(\d+))?", output)
```

Reply:
I guess AI can't parse regular languages when it's trained in a natural one (😏 )